twostep-container-build/README.md

```diff
@@ -0,0 +1,189 @@
+    check_label-->|No|build_image["Build image<br>& push (optional)"]
+    tag_exists-->|No|build_image
+    build_image-->done
+  end
+```
+
+Caching is done by storing the cache-key as a label in the image (`TWO_STEP_BUILD_CACHE_KEY`); and the build and push process is done using the [docker/build-push-action@v6](https://github.com/docker/build-push-action/tree/v6) action. Users have to explicitly provide the cache key for the first step. For example, if you are dealing with an R package, you can cache the dependencies by passing the key `${{ hashFiles('DESCRIPTION') }}` to the `first-step-cache-key` input. That way, the first step will only be executed if the dependencies change.
+
+## Inputs and Outputs
+
+| Field | Description | Required | Default |
+|-------|-------------|----------|---------|
+| `container-file-1` | Path to the first container file | true | |
+| `container-file-2` | Path to the second container file | true | |
+| `first-step-cache-key` | Cache key for the first step | true | |
+| `image` | Name of the image | true | |
+| `registry` | Registry to push the image to | true |  |
+| `username` | Username for the registry | false |  |
+| `password` | Password for the registry | false |  |
+| `main-branch-name` | Name of the main branch | false | `'main'` |
+| `main-branch-tag` | Tag to use for the main branch | false | `'latest'` |
+
+The following are arguments passed to the [docker/build-push-action@v6](https://github.com/docker/build-push-action/tree/v6) action.
+
+| Field | Description | Required | Default |
+|-------|-------------|----------|---------|
+| `push-image-1` | Push the image created during the first step | false | `false` |
+| `push-image-2` | Push the image created during the second step | false | `false` |
+| `build-args-1` | Build arguments for the first step | false | |
+| `build-args-2` | Build arguments for the second step | false | |
+| `labels-1` | Labels for the first step | false | |
+| `labels-2` | Labels for the second step | false | |
+
+The action has the following outputs:
+
+| Field | Description |
+|-------|-------------|
+| `tag` | Container tag of the built image |
+| `branch` | Branch name |
+| `summary` | A summary of the action: (`built`, `re-built`, or `cached`) |
+
+
+## Example: using the Azure container registry
+It is possible to log in by providing an explicit azure container registry username and password, but for CFA use we recommend a service principal-mediated login approach on a self-hosted runner. Here is an example, based on a [real workflow](https://github.com/CDCgov/pyrenew-hew/blob/main/.github/workflows/containers.yaml) from the [`pyrenew-hew`](https://github.com/cdcgov/pyrenew-hew) repo. You'll first need to create a valid [creds string](https://github.com/Azure/login?tab=readme-ov-file#creds) for your Azure Service Principal and store it as a repo secret. In this example, we've named the secret `MY_SERVICE_PRINCIPAL_CREDS_STRING`.
```

Comment:
might want to remove this CFA refernce and potentially just drop this azure section?